docs/content/direct/user-guide.md

```diff
@@ -31,3 +31,59 @@ Additionally, the recipe for registering a WEC with the ITS needs to be modified
+
+## Rule-based customization
+
+KubeStellar can distribute one workload object to multiple WECs, and it is not uncommon for users to need some customization to each WEC. By _rule based_ we mean that the customization is not expressed via one or more literal expressions but rather can refer to _properties_ of each WEC by property name. As KubeStellar distributes or transports a workload object from WDS to a WEC, the object can be transformed in a way that depends on those properties.
+
+KubeStellar currently has a simple but limited way to specify rule-based customization, called "template expansion". This works with templates as defined in the "text/template" standard package of Go, and is applied to each leaf string of an object independently. More expressive customization will be developed in the future.
+
+### Template Expansion
+
+Template expansion is an optional feature that a user can request on an object-by-object basis. The way to request this feature on an object is to put the following annotation on the object.
+
+```yaml
+    control.kubestellar.io/expand-templates: "true"
+```
+
+The customization that template expansion does when distributing an object from a WDS to a WEC is applied independently to each leaf string of the object. The string is first parsed as a template, and then replaced with the result of expanding the template (using the `Execute` method of a `*Template`) using a `map` holding key-value pairs conveying properties of the WEC. For the purposes of template expansion, the "properties" of a WEC are defined by the labels and annotations of the inventory object (`ManagedCluster`) representing that WEC in the inventory. Labels take precedence over annotations. Errors from this process are reported in the status field of the Binding object involved.
+
+Template expansion can only be applied when and where the un-expanded leaf strings pass the validation that the WDS applies, and can only epxress substring replacements.
+
+For example, consider the following example workload object.
+
+```yaml
+apiVersion: logging.openshift.io/v1
+kind: ClusterLogForwarder
+metadata:
+  name: instance
+  namespace: openshift-logging
+  annotations:
+    control.kubestellar.io/expand-templates: "true"
+spec:
+  outputs:
+    - name: remote-loki
+      type: loki
+      url: "https://my.loki.server.com/{{.clustername}}"
+...
+```
+
+The following inventory object provides a value for the `clustername` parameter.
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  annotations:
+    clustername: virgo
+  name: virgo
+...
+```
+
+When distributed to the virgo WEC, that ClusterLogForwarded would say the following.
+
+```yaml
+...
+      url: "https://my.loki.server.com/virgo"
+...
```

Comment:
this example is excellent for user-guide.
it shows the user concrete example for how to use the feature.
no need for much more than that. 
short textual description and concrete example.